src/core/lib/security/credentials/ssl/ssl_credentials.c

```diff
@@ -161,34 +163,49 @@ static void ssl_build_server_config(
 
 grpc_server_credentials *grpc_ssl_server_credentials_create(
     const char *pem_root_certs, grpc_ssl_pem_key_cert_pair *pem_key_cert_pairs,
-    size_t num_key_cert_pairs, int force_client_auth, void *reserved) {
+    size_t num_key_cert_pairs, int force_client_auth,
+    grpc_ssl_server_credentials_create_options *options) {
   return grpc_ssl_server_credentials_create_ex(
       pem_root_certs, pem_key_cert_pairs, num_key_cert_pairs,
       force_client_auth
           ? GRPC_SSL_REQUEST_AND_REQUIRE_CLIENT_CERTIFICATE_AND_VERIFY
           : GRPC_SSL_DONT_REQUEST_CLIENT_CERTIFICATE,
-      reserved);
+      options);
 }
 
 grpc_server_credentials *grpc_ssl_server_credentials_create_ex(
     const char *pem_root_certs, grpc_ssl_pem_key_cert_pair *pem_key_cert_pairs,
     size_t num_key_cert_pairs,
     grpc_ssl_client_certificate_request_type client_certificate_request,
-    void *reserved) {
+    grpc_ssl_server_credentials_create_options *options) {
   grpc_ssl_server_credentials *c =
       gpr_zalloc(sizeof(grpc_ssl_server_credentials));
   GRPC_API_TRACE(
       "grpc_ssl_server_credentials_create_ex("
       "pem_root_certs=%s, pem_key_cert_pairs=%p, num_key_cert_pairs=%lu, "
-      "client_certificate_request=%d, reserved=%p)",
+      "client_certificate_request=%d, options=%p)",
       5, (pem_root_certs, pem_key_cert_pairs, (unsigned long)num_key_cert_pairs,
-          client_certificate_request, reserved));
-  GPR_ASSERT(reserved == NULL);
+          client_certificate_request, options));
   c->base.type = GRPC_CHANNEL_CREDENTIALS_TYPE_SSL;
   gpr_ref_init(&c->base.refcount, 1);
   c->base.vtable = &ssl_server_vtable;
   ssl_build_server_config(pem_root_certs, pem_key_cert_pairs,
                           num_key_cert_pairs, client_certificate_request,
                           &c->config);
+  if (options != NULL) {
+    GRPC_API_TRACE(
+        "grpc_ssl_server_credentials_create_options: "
+        "get_server_credentials_cb = %p, get_server_credentials_cb_arg = %p",
+        2, (options->get_server_credentials_cb,
+            options->get_server_credentials_cb_arg));
+    if (options->get_server_credentials_cb != NULL) {
```

Comment:
nit: the `!= NULL` check is not necessary

Reply:
True, however most of the codebase prefers to be explicit, so I'd prefer to follow the same pattern here.

Reply:
hmm i see `c->base.get_server_credentials_cb = get_server_credentials_cb;` etc. in `grpc_ssl_server_security_connector_create()` though?

Reply:
done.